### PR TITLE
Deletion of certain sections from a pattern

### PR DESCRIPTION
**Pull request by ChristophOt**
The new function delete_range was added in pattern.py. To test it, another function test_delete_range was implemented in test_pattern.py.

### CODE REVIEW

File: xypattern/pattern.py [hunk 1]
```diff
@@ -439,22 +439,37 @@ def from_dict(json_dict: dict) -> Pattern:
...
 
-    def delete_range(self, x_range): 
+    def delete_range(self, x_range: tuple) -> Pattern:
         """
         Deletes data points from the pattern within the given range.
 
         :param x_range: Tuple of two floats of x values, 
-         The data points closest to these two x values remain present
-         in the pattern, but the data points in between them are deleted 
+         The data points within the tuple is deleted
```
Comment (CPrescher) [inline xypattern/pattern.py]: This could be more clear, like  "between x_range[0] and x_range[1]"

File: tests/test_pattern.py [hunk 2]
```diff
@@ -367,6 +367,20 @@ def test_str_representation():
 
 def test_delete_range():
     pattern = Pattern(np.arange(11), np.arange(11), name = 'test')
-    pattern.delete_range([2.3, 7.9])
+    pattern = pattern.delete_range([2.3, 7.9])
```
Comment (CPrescher) [inline tests/test_pattern.py:370]: in the function you state a tuple, but test it with a list

File: xypattern/pattern.py [hunk 3]
```diff
@@ -439,22 +439,37 @@ def from_dict(json_dict: dict) -> Pattern:
...
+    
+    def delete_ranges(self, x_ranges: list) -> Pattern:
+        """
+        Deletes data points from the pattern within each 
+        of the given ranges.
+
+        :param x_ranges: List containing tuples of floats of x values, 
+         The data points within each of the tuples is deleted 
+         from the pattern.
+        :return: New pattern
```
Comment (CPrescher) [inline xypattern/pattern.py]: What new pattern? Specify.

File: xypattern/pattern.py [hunk 4]
```diff
@@ -439,22 +439,37 @@ def from_dict(json_dict: dict) -> Pattern:
...
         """
         x, y = self.data
-        ind_min = np.argmin(np.abs(x - x_range[0]))
-        ind_max = np.argmin(np.abs(x - x_range[1]))
-        ind = np.where((x <= x[ind_min]) | (x >= x[ind_max]))
-        self._pattern_x = x[ind]
-        self._pattern_y = y[ind]
-        self.changed.emit()
+        ind = np.where((x < x_range[0]) | (x > x_range[1]))
```
Comment (CPrescher) [inline xypattern/pattern.py:470]: That means in the current implementation, points which lie on a limit, will be deleted. This is different than you wanted it in the morning :)

Comment (ChristophOt) [reply xypattern/pattern.py:470]: In the old implementation, the data point closest to a provided x value remained in the pattern. It did not matter whether the provided x value was smaller or larger than the closest data point. This does matter now in the new implementation.

**Comment by CPrescher** [comment]
lgtm

**Pull request merged**